jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -753,4 +768,29 @@ public ChannelWithAddress newChannelWithAddress(ClientConnector clientConnector,
             };
         }
     }
+
+    public interface ConnectListener extends EventListener
+    {
+        public void onConnectBegin(SocketChannel s, SocketAddress a);
+
+        public void onConnectSuccess(SocketChannel s);
+
+        public void onConnectFailure(SocketChannel s, Throwable x);
+    }
+
+    @Override
+    public boolean addEventListener(EventListener listener)
+    {
+        if (listener instanceof ConnectListener connectListener)
+            return listeners.add(connectListener);
+        return false;
```

Comment:
No, you need to call `super` here.

Reply:
We decided to move the logic of events list to a `ClientConnector` class instead of using `ContainerLifeCycle.addEventListener()` and `ContainerLifeCycle.removeEventListener()`.

Am I wrong?

Reply:
If `listener` is not an instance of `ConnectListener`, you cannot just drop it: you need to call `super`.